### PR TITLE
CORTX-28890: Validate hostname of data pod

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- To Validate hostname of data pod

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/setup/rgw.py [hunk 1]
```diff
@@ -446,6 +446,10 @@ def _generate_ssl_cert(conf: MappedConf):
     def _update_hax_endpoint(conf: MappedConf, data_pod_hostname: str):
         """Update hax endpoint values in rgw config file."""
         Log.info('Reading motr_ha_endpoint from data pod')
+
+        if not data_pod_hostname:
```
Comment (cdeshmukh) [inline src/setup/rgw.py:450]: _update_hax_endpoint_and_create_admin() should catch this exception and avoid using it further operation, basically skip that iteration

Comment (selvakumaar5496) [reply src/setup/rgw.py:450]: Done..

File: src/setup/rgw.py [hunk 2]
```diff
@@ -537,7 +544,10 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
                     Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
                 data_pod_hostnames.remove(current_data_node)
                 for data_pod_hostname in data_pod_hostnames:
-                    Rgw._update_hax_endpoint(conf, data_pod_hostname)
+                    try:
+                        Rgw._update_hax_endpoint(conf, data_pod_hostname)
+                    except Exception:
+                        break
```
Comment (cdeshmukh) [inline src/setup/rgw.py:550]: shouldnt 'continue' be there if exception occurs while getting hax endpoint

File: src/setup/rgw.py [hunk 3]
```diff
@@ -473,7 +477,10 @@ def _update_hax_endpoint(conf: MappedConf, data_pod_hostname: str):
     def _update_hax_endpoint_and_create_admin(conf: MappedConf):
         """Update motr_ha(hax) endpoint values to rgw config file."""
         current_data_node = socket.gethostname().replace('server', 'data')
-        Rgw._update_hax_endpoint(conf, current_data_node)
+        try:
+            Rgw._update_hax_endpoint(conf, current_data_node)
+        except Exception:
+            pass
```
Comment (cdeshmukh) [inline src/setup/rgw.py:483]: if exception occurs, logic should jump to loop trying out all hax endpoints

File: src/setup/rgw.py (hunk 2 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py:550]: continue will go ahead with creating rgw user.

Comment (cdeshmukh) [reply src/setup/rgw.py:550]: 'continue' will go back to start of the 'for' loop, 'break' will infact deny further iterations with remaining data pods in the list

Comment (selvakumaar5496) [reply src/setup/rgw.py]: My bad, little confused.. I had continue only in the beginning. 
Applied the changes

**Review by shriya-deshmukh — APPROVED** [review]

File: src/setup/rgw.py (hunk 3 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py]: done

**Review by sachitanands — APPROVED** [review]
looks good

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**